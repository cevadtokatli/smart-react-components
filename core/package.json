{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
