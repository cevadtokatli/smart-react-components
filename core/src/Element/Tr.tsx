import styled from 'styled-components'
import applyStyledProps, { StyledProps } from '../styled-props'

export default styled.tr<StyledProps>(applyStyledProps)
