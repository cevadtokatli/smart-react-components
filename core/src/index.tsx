export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
