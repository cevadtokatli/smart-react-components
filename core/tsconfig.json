{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
