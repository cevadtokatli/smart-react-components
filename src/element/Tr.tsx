import styled from "styled-components"
import {Props, attrs, css} from "./"

export default styled.tr.attrs<Props>(attrs)<Props>(css)