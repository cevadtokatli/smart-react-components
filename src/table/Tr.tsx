import styled from "styled-components"
import ElementProps from "../element/Props"
import Tr from "../element/Tr"

export interface Props extends ElementProps {
    type?: string
}

export default styled(Tr)<Props>`
    ${({theme,type}) => type && `
        & > * {
            border-color: ${theme.src.type[type].table.border} !important;
            background: ${theme.src.type[type].table.background} !important;
            color: ${theme.src.type[type].table.font} !important; 
        }

        thead > & > *:before {
            content: "";
            position: absolute;
            left: 0;
            right: 0;
            top: 0;
            bottom: 0;
            z-index: -1;
            background: ${theme.src.type[type].table.striped} !important;
        }
    `}
`