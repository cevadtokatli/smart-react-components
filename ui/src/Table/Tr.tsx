import Tr from '@smart-react-components/core/Element/Tr'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp } from '@smart-react-components/core/types'
import styled from 'styled-components'

export interface Props extends StyledProps {
  hasHover?: boolean
  isEven?: boolean
  isSoft?: boolean
  palette?: PaletteProp
  startIndex?: number
}

export default styled(Tr)<Props>(({ theme, hasHover, isEven, isSoft, palette, startIndex = 0 }) => `
  ${palette
    ? `
      > *:nth-child(n+${startIndex}) {
        ${!isSoft
          ? `
            background: ${!isEven ? theme.$.palette[palette].background : theme.$.palette[palette].dynamic};
            border-color: ${theme.$.palette[palette].dynamicest};
            color: ${theme.$.palette[palette].font};
          `
          : `
            background: ${!isEven ? theme.$.palette[palette].softBackground : theme.$.palette[palette].softDark};
            border-color: ${theme.$.palette[palette].softDarkest};
            color: ${theme.$.palette[palette].softFont};
          `
        }
      }

      ${hasHover
        ? `
          &:hover > *:nth-child(n+${startIndex}) {
            background: ${!isSoft ? theme.$.palette[palette].dynamicer : theme.$.palette[palette].softDarker};
          }
        `
        : ''
      }
    `
    : ''
  }
`)
