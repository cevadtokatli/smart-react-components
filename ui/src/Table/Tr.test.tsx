import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import Tr from './Tr'

describe('<Tr />', () => {
  it('should render component', () => {
    const { asFragment } = render(wrapTheme(
      <table>
        <tbody>
          <Tr  palette="light" hasHover />
        </tbody>
      </table>
    ))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component with soft colors', () => {
    const { asFragment } = render(wrapTheme(
      <table>
        <tbody>
          <Tr  palette="light" hasHover isSoft />
        </tbody>
      </table>
    ))
    expect(asFragment()).toMatchSnapshot()
  })
})
