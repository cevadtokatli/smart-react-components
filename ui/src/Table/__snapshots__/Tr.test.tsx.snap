// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tr /> should render component 1`] = `
<DocumentFragment>
  .c0 > *:nth-child(n+0) {
  background: #fff;
  border-color: #A8C0CD;
  color: #323a46;
}

.c0:hover > *:nth-child(n+0) {
  background: #BBCED8;
}

<table>
    <tbody>
      <tr
        class=" c0"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<Tr /> should render component with soft colors 1`] = `
<DocumentFragment>
  .c0 > *:nth-child(n+0) {
  background: undefined;
  border-color: #f8fafb;
  color: #909394;
}

.c0:hover > *:nth-child(n+0) {
  background: #fafbfc;
}

<table>
    <tbody>
      <tr
        class=" c0"
      />
    </tbody>
  </table>
</DocumentFragment>
`;
